echo-model/src/main/java/com/netflix/spinnaker/echo/model/Trigger.java

```diff
@@ -133,16 +134,21 @@ public Trigger atTag(final String tag) {
         .build();
   }
 
-  public Trigger atPayloadConstraints(final Map parameters, final Map payloadConstraints) {
+  public Trigger atPayload(final Map payload) {
+    return this.toBuilder()
+      .buildNumber(null)
```

Comment:
Shouldn't these only set the necessary fields? This might delete stuff especially when these are chained together.

Reply:
I initially preserved the `null`s because the first at* functions had them, but I agree. I'll see where we can reasonably get rid of these.